pkg/chunked/storage_linux.go

```diff
@@ -654,10 +674,19 @@ func (d *chunkedZstdDiffer) ApplyDiff(dest string, options *archive.TarOptions)
 		return output, err
 	}
 
-	enableHostDedup := false
-	if value := storeOpts.PullOptions["enable_host_deduplication"]; strings.ToLower(value) == "true" {
+	const trueVal = "true"
+
+	if value := storeOpts.PullOptions["enable_partial_images"]; strings.ToLower(value) != trueVal {
+		return output, errors.New("enable_partial_images not configured")
+	}
+
+	enableHostDedup, useHardLinks := false, false
+	if value := storeOpts.PullOptions["enable_host_deduplication"]; strings.ToLower(value) == trueVal {
 		enableHostDedup = true
 	}
+	if value := storeOpts.PullOptions["use_hard_links"]; strings.ToLower(value) == trueVal {
```

Comment:
Three copies seem enough for me to extract this into a helper function, `parseBooleanPullOption(storeOpts, option_name, default)` or something like that.